### PR TITLE
Allow Fetch of chunks containing a  multiple of vector sizes for the Arrow Record Batch Reader

### PR DESCRIPTION
**Pull request by pdet**
Examples: 
```python
import duckdb

duckdb_cursor = duckdb.connect()
duckdb_cursor.execute("CREATE table t as select range a from range(5000);")
query = duckdb_cursor.execute("SELECT a FROM t")
record_batch_reader = query.fetch_record_batch(2)
chunk = record_batch_reader.read_next_batch()
# Our batches are now up to 2 * STANDARD_VECTOR_SIZE
assert(len(chunk) == 2048)
```

```r
con <- dbConnect(duckdb::duckdb())
dbExecute(con, paste0("CREATE table t as select range a from range(5000);"))
res <- dbSendQuery(con, "SELECT * FROM t", arrow=TRUE)
record_batch_reader <- duckdb::duckdb_fetch_record_batch(res,2)
cur_batch <- record_batch_reader$read_next_batch()
# Our batches are now up to 2 * STANDARD_VECTOR_SIZE
expect_equal(2048,cur_batch$num_rows)
```

@jonkeane , let me know if you have any feedback on those, I'm happy to accommodate any changes :-)

### CODE REVIEW

**Comment by jonkeane** [comment]
Hello! This worked, I haven't had a chance to do the performance testing I was hoping to, but it doesn't crash anymore, which is great! Thanks

**Comment by pdet** [comment]
@Mytherin any thought about this PR?

File: tools/pythonpkg/src/pyconnection.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ void DuckDBPyConnection::Initialize(py::handle &m) {
 	         "Fetch a chunk of the result as an Arrow Table following execute()", py::arg("vectors_per_chunk") = 1,
 	         py::arg("return_table") = false)
 	    .def("fetch_record_batch", &DuckDBPyConnection::FetchRecordBatchReader,
-	         "Fetch an Arrow RecordBatchReader following execute()")
+	         "Fetch an Arrow RecordBatchReader following execute()", py::arg("vectors_per_chunk") = 1)
```
Comment (Mytherin) [inline tools/pythonpkg/src/pyconnection.cpp:53]: `vectors_per_chunk` seems like an unclear parameter for a user, and is also not guaranteed to contain a specific number of tuples (e.g. a vector might contain less than 1024 tuples in some cases). Perhaps we could replace this with `batch_size`?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good. One comment:

File: tools/pythonpkg/src/pyconnection.cpp (hunk 1 above)
Comment (Mytherin) [reply tools/pythonpkg/src/pyconnection.cpp:53]: Perhaps an easy compromise is to change this to `approx_batch_size` and then just divide that by the vector size, so `approx_batch_size` of 1024 becomes `vectors_per_chunk=1` internally, etc.

Comment (pdet) [reply tools/pythonpkg/src/pyconnection.cpp:53]: I'm fine with that.

**Review by Mytherin — APPROVED** [review]

**Pull request merged**